mautic/mautic
# Fix api stage and owner fields are kept

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ yes ]
| New feature/enhancement? (use the a.x branch)      | [ no ]
| Deprecations?                          | [ no ]
| BC breaks? (use the c.x branch)        | [ no ]
| Automated tests included?              | [ yes ] 
| Related user documentation PR URL      | none
| Related developer documentation PR URL | 
| Issue(s) addressed                     | Fixes #12072


#### Description:

See Issue: https://github.com/mautic/mautic/issues/12072

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a new contact with firstname, lastname, email (name@mautic.com), stage and owner
3. Change the firstname via the API endpoint /contacts/new
E.g.:

```
curl 'https://mautic.ddev.site/index_dev.php/api/contacts/new' \
--header 'Content-Type: application/json' \
--header 'Authorization: Basic ****' \
--data-raw '{
    "email": "name@mautic.com",
    "firstname": "test"
}'
```

4. Refresh the contact with the email name@mautic.com 
5. The contact has **NOT** lost the assigned stage and the owner
